src/main/java/my/auction/email/service/MailServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package my.auction.email.service;
+
+import java.time.LocalDateTime;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import my.auction.global.util.RedisUtil;
+import my.auction.user.service.UserService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.MailException;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class MailServiceImpl implements MailService {
+
+	//* TODO : Custom Exception 클래스 구현하기
+	
+	private final JavaMailSender javaMailSender;
+	private final RedisUtil redisUtil;
+	private final UserService userService;
+	
+	@Value("${spring.mail.sender}")
+	private String senderMail;
+	
+	private static final long EXPIRE_TIME = 5; // 5분
+	private static final long VERIFIED_EXPIRE_TIME = 60; // 인증 완료 정보는 60분 유지
+	
+	@Override
+	public void sendEmail(String to, String code) {
+		String subject = "[My Auction] 실시간 경매 서비스 이메일 인증";
+		String codeMessage = "인증번호는 : [" + code + "] 입니다. 타인에게 절대 알려주지 마세요.";
+		String message = "안녕하세요 🙂\n\n 본인 인증을 위해 아래 인증번호를 입력해주세요.\n\n" + codeMessage;
+		
+		sendMessage(to, subject, message);
+	}
+	
+	@Override
+	public void sendVerificationCode(String email) {
+		try {
+			if (userService.isUserExists(email)) {
+				if (userService.isEmailVerified(email)) {
+					throw new Exception("이미 인증된 이메일입니다. " + email);
+				}
+			}
```

Comment:
if문이 중첩되면 가독성이 낮아집니다
```suggestion
			if (userService.isUserExists(email) && userService.isEmailVerified(email)) {
				throw new Exception("이미 인증된 이메일입니다. " + email);
			}
```